docs/docs/for-developers/contributing.md

```diff
@@ -24,6 +24,65 @@ To improve the chances to get a pull request merged, you should select an issue
 
 In addition, Onivim 2 is built on [Revery](https://github.com/revery-ui/revery) - any work or improvements there will directly improve Onivim 2, as well!
 
+#### Changelog generation
+
+We use a script to generate the change log based on metadata picked up from the commit message, PR title and PR body. For the changelog to be generated successfully it's therefore important that these follow the exact format explained below.
+
+##### PR title
+
+PR titles should have the following format:
+
+```
+<type>(<scope>): <subject>
+```
+
+where `type` can be one of
+
+* **feat:** A new feature
+* **fix:** A bug fix
+* **refactor:** A code change that neither fixes a bug nor adds a feature
+* **perf:** A code change that improves performance
+* **test:** Adding missing or correcting existing tests
+* **docs:** Documentation only changes
+* **chore:** Changes to the build process or auxiliary tools and libraries such as documentation generation
+* **style:** Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
+
```

Comment:
```suggestion
* **docs:** Documentation only changes
* **chore:** Changes to the build process or auxiliary tools and libraries such as documentation generation

```

I don't think this is needed, as we check this in CI anyway. And if for some reason we do on occasion get something like this, we can just do it as a chore I think.